src/pcp/buddyinfo/pcp-buddyinfo.py

```diff
@@ -161,22 +167,47 @@ def report(self, manager):
 
 class BuddyinfoOptions(pmapi.pmOptions):
     timefmt = "%H:%M:%S"
+    uflag = False
+
+    def extraOptions(self,opt,optarg,index):
+        if opt == 'u':
+            BuddyinfoOptions.uflag = True
+
     def __init__(self):
-        pmapi.pmOptions.__init__(self, "a:s:Z:zV?")
+        pmapi.pmOptions.__init__(self, "a:s:Z:uzV?")
         self.pmSetLongOptionHeader("General options")
+        self.pmSetOptionCallback(self.extraOptions)
+        self.pmSetLongOptionArchive()
+        self.pmSetLongOption("no-interpolation", 0, "u", "", "disable interpolation mode with archives")
```

Comment:
@orasagar could you make the long-option here match with pmrep too?  (--no-interpol)

Reply:
done